qp_klp/Step.py

```diff
@@ -369,32 +386,140 @@ def generate_sifs(self, from_qiita):
-                for prep_file in prep_info_file_paths:
-                    # not the best check but good enough for now.
-                    if proj in prep_file:
-                        df = pd.read_csv(prep_file, sep='\t',
-                                         dtype=str, index_col=False)
-                        # save copy of sample_name column as 'old_sample_name'
-                        df['old_sample_name'] = df['sample_name']
-                        for i in df.index:
-                            smpl_name = df.at[i, "sample_name"]
-                            if not smpl_name.startswith('BLANK'):
-                                # remove any leading zeroes if they exist
-                                smpl_name = smpl_name.lstrip('0')
-                                if smpl_name in sn_tid_map_by_proj[proj]:
-                                    tube_id = sn_tid_map_by_proj[proj][
-                                        smpl_name]
-                                    df.at[i, "sample_name"] = tube_id
-                        df.to_csv(prep_file, index=False, sep="\t")
-
-    def update_blanks_in_qiita(self, sifs, qclient):
-        for sif_path in sifs:
+    def get_tube_ids_from_qiita(self, qclient):
+        # Update get_project_info() so that it can return a list of
+        # samples in projects['samples']. Include blanks in projects['blanks']
+        # just in case there are duplicate qiita_ids
+        qiita_ids = [x['qiita_id'] for x in
+                     self.pipeline.get_project_info(short_names=True)]
+
+        tids_by_qiita_id = {}
+        sample_names_by_qiita_id = {}
+
+        for qiita_id in qiita_ids:
+            # Qiita returns a set of sample-ids in qsam and a dictionary where
+            # sample-names are used as keys and tube-ids are their values.
+            qsam, tids = self.get_samples_in_qiita(qclient, qiita_id)
+
+            if tids is None:
+                sample_names_by_qiita_id[str(qiita_id)] = qsam
+            else:
+                # fix values in tids to be a string instead of a list of one.
+                # also, remove the qiita_id prepending each sample-name.
+                tids = {k.replace(f'{qiita_id}.', ''): tids[k][0] for k in
+                        tids}
+
+                # the values Qiita returns for tids seems like it can include
+                # empty strings if there is no tube-id associated with a
+                # sample-name. For now assume it doesn't happen in production
+                # and if prep-files have empty sample-names we'll know.
+                tids_by_qiita_id[str(qiita_id)] = tids
+
+        # use empty dict {} as an indication that get_tube_ids_from_qiita was
+        # called but no tube-ids were found for any project.
+        self.tube_id_map = tids_by_qiita_id
+        self.samples_in_qiita = sample_names_by_qiita_id
+
+        # not needed, but useful in displaying debugging info
+        return tids_by_qiita_id
```

Comment:
If not needed, please rm (similar to my previous comment) ... you can always use self.tube_id_map for debugging, right?